ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx

```diff
@@ -33,28 +32,18 @@ const OperationIcon = ({ operation }: OperationComponentProps) => {
   switch (operation.operationType) {
     case OnboardingOperationTypeEnum.ONBOARDING:
       return <Icon name={"ok"} />;
-
     case IbanOperationTypeEnum.ADD_IBAN:
       return <Icon name={"institution"} color="bluegreyLight" />;
-
     case RefundOperationTypeEnum.PAID_REFUND:
       return <Icon name="refund" color="bluegrey" />;
-
-    case OperationTypeEnum.REJECTED_ADD_INSTRUMENT:
-    case OperationTypeEnum.REJECTED_DELETE_INSTRUMENT:
+    case RejectedInstrumentOperationTypeEnum.REJECTED_ADD_INSTRUMENT:
+    case RejectedInstrumentOperationTypeEnum.REJECTED_DELETE_INSTRUMENT:
     case RefundOperationTypeEnum.REJECTED_REFUND:
       return <Icon name={"legWarning"} color="red" />;
-
+    case TransactionOperationTypeEnum.REVERSAL:
+    case TransactionOperationTypeEnum.TRANSACTION:
+      return getCardLogoComponent(operation.brand);
     default:
```

Comment:
why did you remove the generic credit card icon in case of malformed data here?

Reply:
I took the chance to refactor this piece of code. I replaced `instrumentBrandMap`, which contained incomplete data, with the `getCardLogoComponent`. Now it is aligned with the rest of the app.

In case you missed it, `getCardLogoComponent` is an utility function that does everything necessary to display the correct card brand logo:
- It checks if contains a valid `IOPaymentLogo` string.
- If not, it checks if the string was sent in uppercase format.
- If all the previous checks fail, it returns the generic credit card icon.
- If the data is correct, it returns the correct `LogoPayment`.